indra/llrender/llfontgl.cpp

```diff
@@ -341,24 +349,35 @@ S32 LLFontGL::render(const LLWString &wstr, S32 begin_offset, F32 x, F32 y, cons
                     (F32)ll_round(cur_render_x + (F32)fgi->mXBearing) + (F32)fgi->mWidth,
                     (F32)ll_round(cur_render_y + (F32)fgi->mYBearing) - (F32)fgi->mHeight);
 
-        draw_queued_glyphs();
+        // Prevent the buffer overflow
+        render_queued_glyphs(GLYPH_BATCH_SIZE - glyph_per_char);
 
         const LLColor4U& col =
             bitmap_entry.first == EFontGlyphType::Grayscale ? text_color
                                                             : emoji_color;
+        // This function emplaces data to arrays vertices, uvs, colors
+        // and increases glyph_count on 1, 2 or 6 depending on shadow
         drawGlyph(glyph_count, vertices, uvs, colors, screen_rect, uv_rect,
                   col, style_to_add, shadow, drop_shadow_strength);
+        llassert(glyph_count <= GLYPH_BATCH_SIZE);
+        if (!glyph_per_char)
+        {
+            glyph_per_char = glyph_count;
```

Comment:
Shouldn't this be drawGlyph(glyph_per_char, ... and glyph_count += glyph_per_char;?

Reply:
Definitely No

Each call of `drawGlyph` increases `glyph_count` on the same value (1, 2 or 6, depending on `shadow`)
Initially `glyph_per_char` is set to 0, and only after the first call of `drawGlyph` it is reinitialized again
Using this value we can prognose the next size of `glyph_count` before the next call of `drawGlyph`

Reply:
Aren't you missing a reset in such case?

Reply:
Yes, the parameter `shadow` is the same for all characters in the provided string,
so the value of `glyph_per_char` is absolutely the same for the whole string fragment